godtools/App/Features/ToolsFilter/Presentation/ToolFilterCategorySelectionView/ToolFilterCategorySelectionViewModel.swift

```diff
@@ -43,22 +45,37 @@ class ToolFilterCategorySelectionViewModel: ObservableObject {
         self.selectedCategory = categoryFilterSelectionPublisher.value
         self.flowDelegate = flowDelegate
         
+        getCurrentAppLanguageUseCase
+            .getLanguagePublisher()
+            .assign(to: &$appLanguage)
+        
+        $appLanguage.eraseToAnyPublisher()
+            .flatMap { appLanguage in
+                
+                return viewToolFilterCategoriesUseCase.viewPublisher(filteredByLanguageId: selectedLanguage.id, translatedInAppLanguage: appLanguage)
+            }
+            .receive(on: DispatchQueue.main)
+            .sink { viewCategoryFiltersDomainModel in
+                
+                self.allCategories = viewCategoryFiltersDomainModel.categoryFilters
+            }
+            .store(in: &cancellables)
+        
         getInterfaceStringInAppLanguageUseCase
             .getStringPublisher(id: ToolStringKeys.ToolFilter.categoryFilterNavTitle.rawValue)
             .receive(on: DispatchQueue.main)
             .assign(to: &$navTitle)
         
-        getToolFilterCategoriesUseCase.getToolFilterCategoriesPublisher(filteredByLanguageId: selectedLanguage.id)
-            .receive(on: DispatchQueue.main)
-            .assign(to: &$allCategories)
-        
-        searchToolFilterCategoriesUseCase
-            .getSearchResultsPublisher(
-                for: $searchText.eraseToAnyPublisher(),
-                in: $allCategories.eraseToAnyPublisher()
-            )
-            .receive(on: DispatchQueue.main)
-            .assign(to: &$categorySearchResults)
+        Publishers.CombineLatest(
+            $searchText.eraseToAnyPublisher(),
+            $allCategories.eraseToAnyPublisher()
+        )
+        .flatMap { searchText, allCategories in
+            
+            self.searchToolFilterCategoriesUseCase.getSearchResultsPublisher(for: searchText, in: allCategories)
```

Comment:
Capturing self here in flatMap will cause a retain cycle. Easily resolved by using searchToolFilterCategoriesUseCase that is in the scope of init.